plugins/auto-sizes/optimization-detective.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Optimization Detective extensions by Auto Sizes.
+ *
+ * @since n.e.x.t
+ * @package auto-sizes
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Visits responsive lazy-loaded IMG tags to ensure they include sizes=auto.
+ *
+ * @since n.e.x.t
+ *
+ * @param OD_Tag_Visitor_Context $context Tag visitor context.
+ * @return false Whether the tag should be recorded in URL metrics.
+ */
+function auto_sizes_visit_tag( OD_Tag_Visitor_Context $context ): bool {
+	if ( 'IMG' !== $context->processor->get_tag() ) {
+		return false;
+	}
+
+	$sizes = $context->processor->get_attribute( 'sizes' );
+	if ( ! is_string( $sizes ) || 'lazy' !== $context->processor->get_attribute( 'loading' ) ) {
+		return false;
```

Comment:
Oh, what if `sizes=auto` had been added by Auto Sizes but Image Prioritizer removed `loading=lazy` since the image actually appears in the initial viewport? Should this make sure that `auto` is removed from `sizes` as well? Is it bad if a non-lazy loaded image includes `sizes=auto`?

Reply:
Addressed in f5e4231c